### PR TITLE
docs: async-js

### PR DESCRIPTION
**Pull request by Sciator**
Closes https://github.com/influxdata/influxdb-client-js/issues/610

Changes for [nodejs query](https://docs.influxdata.com/influxdb/v2.4/api-guide/client-libraries/nodejs/query/)
Screenshot 2022-10-23 at 13 35 43
Changes for [nodejs query Complete example](https://docs.influxdata.com/influxdb/v2.4/api-guide/client-libraries/nodejs/query/#complete-example)
Screenshot 2022-10-23 at 13 36 53

changes are done for v2.4-v2.0

- [x] Signed the [InfluxData CLA](https://www.influxdata.com/legal/cla/)
- [x] Rebased/mergeable

### CODE REVIEW

File: content/influxdb/v2.0/api-guide/client-libraries/nodejs/query.md [hunk 1]
```diff
@@ -48,33 +48,29 @@ The following example sends a Flux query to an InfluxDB bucket and outputs rows
...
-
-   queryApi.queryRows(fluxQuery, observer)
-
-   ```
+4. Use the `iterateRows()` method of the query client to query InfluxDB.
+   `iterateRows()` takes a Flux query and returns table as an asynchronous collection.
+   The client returns [table](/{{% latest "influxdb" %}}/reference/syntax/annotated-csv/#tables) metadata and rows as an as an AsyncIterable.
+
+  ```js
+  const myQuery = () => {
```
Comment (sranka) [inline content/influxdb/v2.0/api-guide/client-libraries/nodejs/query.md:64]: This should rather be an async lambda with proper error handling inside

File: content/influxdb/v2.0/api-guide/client-libraries/nodejs/query.md [hunk 2]
```diff
@@ -48,33 +48,29 @@ The following example sends a Flux query to an InfluxDB bucket and outputs rows
    To prevent SQL injection attacks, avoid concatenating unsafe user input with queries.
    {{% /warn %}}
 
-   ```js
-   const fluxQuery =
-     'from(bucket: "YOUR_BUCKET")
-     |> range(start: 0)
-     |> filter(fn: (r) => r._measurement == "temperature")'
-   ```
+    ```js
```
Comment (sranka) [inline content/influxdb/v2.0/api-guide/client-libraries/nodejs/query.md:51]: useless change out of scope of this PR, isn't it ?

File: content/influxdb/v2.1/api-guide/client-libraries/nodejs/query.md [hunk 3]
```diff
@@ -48,33 +48,29 @@ The following example sends a Flux query to an InfluxDB bucket and outputs rows
...
-
-   queryApi.queryRows(fluxQuery, observer)
-
-   ```
+4. Use the `iterateRows()` method of the query client to query InfluxDB.
+   `iterateRows()` takes a Flux query and returns table as an asynchronous collection.
+   The client returns [table](/{{% latest "influxdb" %}}/reference/syntax/annotated-csv/#tables) metadata and rows as an as an AsyncIterable.
+
+  ```js
+  const myQuery = () => {
```
Comment (sranka) [inline content/influxdb/v2.1/api-guide/client-libraries/nodejs/query.md:64]: This should an async lambda with proper error handling, don't you think?

File: shared/text/api/v2.0/query/query.mjs [hunk 4]
```diff
@@ -21,21 +21,14 @@ const queryApi = new InfluxDB({url, token}).getQueryApi(org)
 /** To avoid SQL injection, use a string literal for the query. */
 const fluxQuery = 'from(bucket:"air_sensor") |> range(start: 0) |> filter(fn: (r) => r._measurement == "temperature")'
 
-const fluxObserver = {
-  next(row, tableMeta) {
-    const o = tableMeta.toObject(row)
+const myQuery = () => {
```
Comment (sranka) [inline shared/text/api/v2.0/query/query.mjs:24]: ... should be an async lambda with proper error handling ?

File: content/influxdb/v2.1/api-guide/client-libraries/nodejs/query.md (hunk 3 above)
Comment (Sciator) [reply content/influxdb/v2.1/api-guide/client-libraries/nodejs/query.md:64]: Users should learn about error handling in general js tutorials, so for this example is error handling just added unnecessary complexity.
Missing async keyword is mistake indeed.

Comment (sranka) [reply content/influxdb/v2.1/api-guide/client-libraries/nodejs/query.md]: Uncaught promise rejection terminates the node.js process, that's why I think it is rather a good practice to let inexperienced users to learn from good examples. I wanted to let know about the facts and let others decide, I am not an approver herein ... also, the previous version was even worse, it was not correct about using observables at all.

File: content/influxdb/v2.0/api-guide/client-libraries/nodejs/query.md [hunk 5]
```diff
@@ -56,24 +56,20 @@ The following example sends a Flux query to an InfluxDB bucket and outputs rows
    ```
    Replace *`YOUR_BUCKET`* with the name of your InfluxDB bucket.
 
-4. Use the `queryRows()` method of the query client to query InfluxDB.
-   `queryRows()` takes a Flux query and an [RxJS **Observer**](http://reactivex.io/rxjs/manual/overview.html#observer) object.
-   The client returns [table](/{{% latest "influxdb" %}}/reference/syntax/annotated-csv/#tables) metadata and rows as an  [RxJS **Observable**](http://reactivex.io/rxjs/manual/overview.html#observable).
-   `queryRows()` subscribes your observer to the observable.
-   Finally, the observer logs the rows from the response to the terminal.
+4. Use the `iterateRows()` method of the query client to query InfluxDB.
+   `iterateRows()` takes a Flux query and returns table as an asynchronous collection.
```
Comment (jstirnaman) [inline content/influxdb/v2.0/api-guide/client-libraries/nodejs/query.md:60]:
```suggestion
   `iterateRows()` takes a Flux query and returns the [table](/{{% latest "influxdb" %}}/reference/syntax/annotated-csv/#tables) of metadata and rows as an asynchronous iterable (`AsyncIterable<Row>`).
```

File: content/influxdb/v2.0/api-guide/client-libraries/nodejs/query.md [hunk 6]
```diff
@@ -56,24 +56,20 @@ The following example sends a Flux query to an InfluxDB bucket and outputs rows
...
+   The client returns [table](/{{% latest "influxdb" %}}/reference/syntax/annotated-csv/#tables) metadata and rows as an as an AsyncIterable.
```
Comment (jstirnaman) [inline content/influxdb/v2.0/api-guide/client-libraries/nodejs/query.md:61]:
```suggestion
   The following example shows how to write an asynchronous function that uses the `iterateRows()` method to query a bucket and uses the JavaScript `for await...of` statement to iterate over the query results:

```

File: content/influxdb/v2.0/api-guide/client-libraries/nodejs/query.md [hunk 7]
```diff
@@ -56,24 +56,20 @@ The following example sends a Flux query to an InfluxDB bucket and outputs rows
...
        console.log(
          `${o._time} ${o._measurement} in '${o.location}' (${o.sensor_id}): ${o._field}=${o._value}`
        )
      }
    }
-
-   queryApi.queryRows(fluxQuery, observer)
-
+   myQuery()
    ```
```
Comment (jstirnaman) [inline content/influxdb/v2.0/api-guide/client-libraries/nodejs/query.md:73]:
```suggestion
   ```

To learn more about `for await...of` and async iterable objects, see the [MDN `for await...of` documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for-await...of).
```

File: content/influxdb/v2.1/api-guide/client-libraries/nodejs/query.md [hunk 8]
```diff
@@ -14,7 +14,7 @@ aliases:
 
 Use the [InfluxDB JavaScript client library](https://github.com/influxdata/influxdb-client-js) in a Node.js environment to query InfluxDB.  
 
-The following example sends a Flux query to an InfluxDB bucket and outputs rows from an observable table.
+The following example sends a Flux query to an InfluxDB bucket and outputs rows as asynchronous iterable.
```
Comment (jstirnaman) [inline content/influxdb/v2.1/api-guide/client-libraries/nodejs/query.md:17]:
```suggestion
The following example sends a Flux query to an InfluxDB bucket and outputs rows as a JavaScript _asynchronous iterable_ object.
```

**Review by jstirnaman — APPROVED** [review]
@Sciator @sranka I'll send a separate PR to cleanup the remaining style, grammar, and linking once the code is complete.

**Pull request merged**